KsApi/models/graphql/adapters/Project+FetchProjectRewardsByIdQueryData.swift

```diff
@@ -13,9 +13,56 @@ extension Project {
 
   static func projectRewards(from data: GraphAPI.FetchProjectRewardsByIdQuery.Data) -> [Reward] {
     let projectRewards = data.project?.rewards?.nodes?
-      .compactMap { $0?.fragments.rewardFragment }
-      .compactMap { Reward.reward(from: $0) } ?? []
+      .compactMap { node -> (GraphAPI.RewardFragment, [ShippingRule]?)? in
+        guard let rewardFragment = node?.fragments.rewardFragment else { return nil }
 
-    return projectRewards
+        // These shipping rules are constructed from simplified versions of the shipping rules.
+        // Names are not localized and should not be shown to users, but they contain the data we
+        // need to calculate shipping, just in a different initial format.
+        let expandedShippingRules = node?.simpleShippingRulesExpanded?
+          .compactMap { node -> ShippingRule? in
+            guard let node,
+                  let idString = node.locationId, let locationId = decompose(id: idString)
+            else {
+              return nil
+            }
+
+            let name = node.locationName ?? ""
```

Comment:
Should the guard statement return nil here if there's no location name?

Reply:
I don't think it needs to, since we're not actually using this name anywhere. (It's not localized, so we're relying on the shipping rules we fetch from v1 to provide us with a name.)